### PR TITLE
Fix/500 targeted architectural polish (part 1)

### LINKED ISSUE #500: Targeted architectural polish (part 1)
Refactoring and polishing of various elements of the codebase:
- model.group 
- model.event 
- model.points

### PR DESCRIPTION
**Pull request by AACioffi**
## What?
Polish work for Sprint 10 covering model classes, repository implementations, and documentation. 

Added comprehensive KDoc to `Group` and `Event` data classes, fixed out of place variable name in `PointsLocalRepository`, implemented four TODO methods in `GroupsLocalRepository` (member/admin management), cleaned up unused code, corrected inaccurate documentation in repository interfaces, and added complete test coverage for `GroupsLocalRepository` (16 tests).

Closes #500 .

## Why?

**Documentation Gaps**: Model classes (`Group`, `Event`) and repository implementations lacked class-level KDoc, making codebase harder to navigate and understand. Repository interfaces had incomplete or misleading documentation.

**Incomplete Implementation**: `GroupsLocalRepository` had four TODO methods for member/admin management that needed implementation for feature completeness.

**Code Quality Issues**:
- Inaccurate variable `events` describing points repository in `PointsLocalRepository`
- Unused code: empty serializers package, unused `EventStatus.displayString()` extension, `pointsSourceToString()` in `Points.kt`
- Generic exception message in `EventsLocalRepository`
- Inconsistent documentation across Firestore repositories

**Scope**: Affects `model.group`, `model.event`, `model.points` packages and all repository layers (local, Firestore, interfaces).

## How?

**Documentation**:
- Added comprehensive KDoc to `Group` and `Event` data classes with `@property` tags for all fields
- Added class-level KDoc to all local repositories explaining their in-memory behaviour
- Added class-level KDoc to all Firestore repositories documenting Firebase-specific behaviour and exceptions
- Removed redundant method-level KDoc from `EventsRepositoryFirestore`
- Fixed misleading documentation in `PointsRepository` ("chronological" --> "sorted by date, newest first")

**Bug Fixes**:
- Renamed `events` --> `pointList` in `PointsLocalRepository`
- Added descriptive message to `IllegalArgumentException` in `EventsLocalRepository`

**Implementation**:
- `GroupsLocalRepository.addMember()`: Validates group exists, creates updated Group with new member, replaces in list via `editGroup()`
- `GroupsLocalRepository.removeMember()`: Validates group exists, creates updated Group without member (also removes from admins if present), replaces in list via `editGroup()`
- `GroupsLocalRepository.addAdmin()`: Validates group exists, creates updated Group with new admin, replaces in list via `editGroup()`
- `GroupsLocalRepository.removeAdmin()`: Validates group exists, creates updated Group without admin, replaces in list via `editGroup()`

**Code Cleanup**:
- Removed unused serializers package
- Removed unused `EventStatus.displayString()` extension function
- Removed unused `pointsSourceToString()` method in `Points.kt`

**Tests**:
- Created `GroupsLocalRepositoryTest` covering all CRUD operations, member management, admin management, and edge cases (16 tests)
- Created `GroupsLocalRepositoryTestData` for test data

## Testing?

**Unit Tests**: `GroupsLocalRepositoryTest`, with 100% pass rate
**Existing Test Suites**: All passing - changes are backward compatible (implementations were not changed, except for `GroupsLocalRepository`, for which the previously non-implemented functions were unused).

## Screenshots

image
All tests in `GroupsLocalRepositoryTest` pass successfully.

image

Line coverage > 80%.

### CODE REVIEW

File: app/src/main/java/com/android/gatherly/model/group/GroupsLocalRepository.kt [hunk 1]
```diff
@@ -53,19 +53,31 @@ class GroupsLocalRepository : GroupsRepository {
...
   }
 
   override suspend fun removeMember(groupId: String, userId: String) {
-    TODO("GroupsLocalRepository.removeMember Not yet implemented")
+    val group = getGroup(groupId)
+    editGroup(
+        groupId,
+        group.copy(
+            memberIds = group.memberIds.filter { it != userId },
+            adminIds = group.adminIds.filter { it != userId }))
```
Comment (clairechaffard) [inline app/src/main/java/com/android/gatherly/model/group/GroupsLocalRepository.kt]: Question: Should you verify that the userId is not the last remaining admin in the group ? If so, should an error exception be thrown ?

Comment (AACioffi) [reply app/src/main/java/com/android/gatherly/model/group/GroupsLocalRepository.kt]: To answer both of the questions you asked, I indicated at the top of the file that the local implementation is simplified and does not behave like the real implementation in all cases.

This is fine in my eyes, because the local implementation is merely used as a support for VM testing, and fulfilled its role in validating VM functionality.
However, I could change it to make it behave more similarly to the real implementation, if you'd prefer.

Comment (clairechaffard) [reply app/src/main/java/com/android/gatherly/model/group/GroupsLocalRepository.kt]: You're right, it's a minor nitpick regarding a detail of the Firebase implementation and it's not important in our case. I will approve your work, it's seems perfect for me.
